File: inbloom-hello-world-c/inBloom-c-sharp-hello-world/Recipes/Recipe5.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using InBloomClient.Entities;
using InBloomClient.Enum;
using System.Net.Http;
using System.Threading.Tasks;
using System.Net;
using RestSharp;

namespace inBloom_c_sharp_hello_world.Recipes
{
    public class Recipe5 : Recipe
    {

        override public string RunRecipe(string token)
        {
            // Create the assessment it's required objects
            AssessmentIdentificationCode idCode = createAssesmentIdentifcationCode(
                identificationSystem: AssessmentIdentificationSystemType.Other,
                ID: "Unique Alphanumberic Identifier"
                );

            List<AssessmentIdentificationCode> idCodes = new List<AssessmentIdentificationCode>();
            idCodes.Add(idCode);

            Assessment assessment = createAssessment(
                assessmentTitle: "Title", 
                assessmentIdentificationCode: idCodes,
                version: 1);

            // return location or error
            return postAssessment(token, assessment);
        }

        private string postAssessment(string token, Assessment assessment)
        {
            var client = new RestClient("https://api.sandbox.inbloom.org/api/rest/v1.2/");
            var endpoint = "assessments/";

            var request = inBloomRestRequest(token, endpoint, Method.POST);
            request.AddBody(assessment);

            RestResponse response = (RestResponse)client.Execute(request);


            // RestSharp recommended error code
            if (response.ErrorException != null)
            {
                const string message = "Error rectrieving response. Check details in exception for more info.";
                var inBLoomException = new ApplicationException(message, response.ErrorException);
                throw inBLoomException;
            }

            if (response.ResponseStatus == ResponseStatus.Completed)
            {
                foreach (Parameter header in response.Headers)
                {
                    if (header.Name == "Location")
                    {
                        // Return the id value of the new assessment
                        string location = header.Value.ToString();
                        string assessmentId = location.Replace(response.ResponseUri.ToString(), "");
                        return assessmentId;
                    }
                }
                // Success but no location -> fail loudly
                const string message = "No location value in completed response. Did the post actually work?";
                var inBLoomException = new ApplicationException(message, response.ErrorException);
                throw inBLoomException;
            }
            else
            {
                return "Post was not successful: " + response.ResponseStatus.ToString();
            }
        }

        private AssessmentIdentificationCode createAssesmentIdentifcationCode(AssessmentIdentificationSystemType identificationSystem, string ID)
        {
            AssessmentIdentificationCode idCode = new AssessmentIdentificationCode();
            idCode.identificationSystem = identificationSystem;
            idCode.ID = ID;

            return idCode;
        }

        private Assessment createAssessment(string assessmentTitle, List<AssessmentIdentificationCode> assessmentIdentificationCode, int version)
        {
            Assessment assessment = new Assessment();
            assessment.assessmentTitle = assessmentTitle;
            assessment.assessmentIdentificationCode = assessmentIdentificationCode;
            //assessment.version = version;

            return assessment;
        }
    }
}

File: inbloom-hello-world-ruby/app/controllers/recipes/recipe5.rb
# Create an assessment
def recipe5()

  assess_loc, new_assessment = create_assessment() 
  
  # Clean up the datastore
  RestClient.delete assess_loc, headers

  
  return assess_loc, new_assessment
end

def create_assessment()
  # create a new assessment onject
  new_assessment = assessment(
    "Generic Assessment Name2", 
    [assessment_identification_code("Other","Some Identification")],
    1
  )
 
  url = REST_URL + 'assessments'
  
  #store the new assesment in the data store
  response = RestClient.post url, new_assessment.to_json, headers
  assess_loc = response.headers[:location]  
 
  return assess_loc, new_assessment.to_json
end

# helper methods
################
def assessment( assessment_title, array_of_assessment_identification_codes, version, optional ={} )
  assessment_hash = {
    :assessmentTitle => assessment_title,
    :assessmentIdentificationCode => array_of_assessment_identification_codes,
    :version => version
  }
  assessment_hash.update optional
end

def assessment_identification_code( identification_system, id, optional ={} )
   assessment_identification_code = {
    :identificationSystem => identification_system,
    :ID => id
   }
   assessment_identification_code.update optional
end
